tests/components/pjlink/test_media_player.py

```diff
@@ -0,0 +1,365 @@
+            mocked_instance.get_inputs.return_value = (("HDMI", 1),)
+            mocked_instance.get_input.return_value = ("HDMI", 1)
+
+        assert await async_setup_component(
+            hass,
+            media_player.DOMAIN,
+            {
+                media_player.DOMAIN: {
+                    "platform": "pjlink",
+                    "host": "127.0.0.1",
+                }
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        state = hass.states.get("media_player.test")
+        assert state.state == "on"
+
+        assert state.attributes["source"] == "HDMI 1"
+
+
+async def test_api_error(projector_from_address, hass):
+    """Test invalid api responses."""
+
+    with assert_setup_component(1, media_player.DOMAIN):
+        instance = projector_from_address.return_value
+
+        with instance as mocked_instance:
+            mocked_instance.get_name.return_value = "Test"
+            mocked_instance.get_inputs.return_value = (
+                ("HDMI", 1),
+                ("HDMI", 2),
+                ("VGA", 1),
+            )
+            mocked_instance.get_power.side_effect = KeyError("OK")
+
+        assert await async_setup_component(
+            hass,
+            media_player.DOMAIN,
+            {
+                media_player.DOMAIN: {
+                    "platform": "pjlink",
+                    "host": "127.0.0.1",
+                }
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        state = hass.states.get("media_player.test")
+        assert state.state == "off"
+
+
+async def test_update_unavailable(projector_from_address, hass):
+    """Test update to a device that is unavailable."""
+
+    with assert_setup_component(1, media_player.DOMAIN):
+        instance = projector_from_address.return_value
+
+        with instance as mocked_instance:
+            mocked_instance.get_name.return_value = "Test"
+            mocked_instance.get_inputs.return_value = (
+                ("HDMI", 1),
+                ("HDMI", 2),
+                ("VGA", 1),
+            )
+
+        assert await async_setup_component(
+            hass,
+            media_player.DOMAIN,
+            {
+                media_player.DOMAIN: {
+                    "platform": "pjlink",
+                    "host": "127.0.0.1",
+                }
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        state = hass.states.get("media_player.test")
+        assert state.state == "off"
+
+        projector_from_address.side_effect = socket.timeout
+        async_fire_time_changed(hass, dt.utcnow() + timedelta(seconds=10))
+        await hass.async_block_till_done()
+
+        state = hass.states.get("media_player.test")
+        assert state.state == "unavailable"
+
+
+async def test_unavailable_time(mocked_projector, hass):
+    """Test unavailable time projector error."""
+
+    assert await async_setup_component(
```

Comment:
A future improvement could be to extract the setup of the integration to a pytest fixture.